### DISCUSSION TITLE
Fragen zum ESP32 S3

### DISCUSSION THREAD

**Discussion by Agerter** [hoylabs/OpenDTU-OnBattery#1091] [General]
Hallo ich bin auch neu auf dem Gebiet und habe mal paar Fragen an die Experten hier.

- Warum blinkt die RGB LED auf dem ESP32 S3 Board wie wild in allen Farben wenn eine Verbindung zum WR besteht.? Kann man das deaktivieren ?
- Warum bekommen ich kein WLAN wenn ich die factory.bin mit flashprogram ESP Downloadtool flashe, bei normalen ESP32 gibt's keine Probleme?. Ein WLAN empfang bekomme ich nur wenn ich das mit dem Webflasher hier flashe. Ich hab auch paar ESP da besteht wohl das bekannte WLAN Problem da gehts mit beiden nicht.

Wenn jemand mir da weiterhelfen kann wäre ich dankbar.

**Comment by Manos1966** [comment]
Hallo @Agerter 

hat vielleicht etwas damit zu tun?
Note: Users report that some board-versions have WiFi problems https://github.com/helgeerbe/OpenDTU-OnBattery/issues/935#issuecomment-2138018534

**Reply by SW-Niko** [reply]
Hallo @Agerter,
hast du dein pin mapping an den S3 angepasst?. Von Espessif gibt es folgende Varianten.

Both the initial and v1.1 versions of ESP32-S3-DevKitC-1 are available on the market. The main difference lies in the GPIO assignment for the RGB LED: the initial version uses GPIO48, whereas v1.1 uses GPIO38.

Und andere GPIOs haben auch unterschiedliche Belegung.

**Reply by Agerter** [reply]
hallo ich habe es über den Webflasher gemacht da ist die Pinmapping dann schon mit drauf

**Reply by schlimmchen** [reply]
Da ist **eine** pin_mapping.json drauf, soweit ich weiß ist es die fürs Fusion-Board.

**Reply by SW-Niko** [reply]
Genau, wenn du irgendein ESP32-S3 Modul gekauft hast und nicht das OpenDTU Fusion dann must du dir ein pin_mapping passend zu deinem Modul erstellen. Das default pin_mapping wird vermutlich nicht passen. Einfach überprüfen und richtig einstellen. Die Anleitung dazu findest du in der Wiki.

**Reply by Agerter** [reply]
die Pinmapping passt vom Webflasher hab die Pins entsprechend angeschlossen das geht...
was nicht geht und das versteh ich nicht ist das WLAN über das flash  SP32 Download tool. wenn ich den ESP32-S3 über den Onlieflasher hier auf github flashe habe ich WLAN auf dem chip.. das versteh ich nicht was ist da anders?

**Reply by Manos1966** [reply]
@Agerter mache bitte ein Bild vom Flash SP32 Download tool und der Dateien (und Position) die sie geladen werden.

Ich darf davon ausgehen, dass du bereits ein "normales ESP32 Board (kein S3) mit dem gleichen  Flash SP32 Download tool  geflasht hast und es hat funktioniert ohne Probleme, richtig?

**Reply by Agerter** [reply]
ja ich hab schon einige normal ESp32 geflashed alles kein Problem, nur eben der ESP S3 bekommt kein WLAN wenn ich mit diesem tool flashe

**Comment by Agerter** [comment]
![flash](https://github.com/user-attachments/assets/4cc75a25-d292-439c-ac6f-017b48eb2433)

**Reply by Agerter** [reply]
so sieht es aus hab den S3 ESp ausgewaehlt

**Reply by Agerter** [reply]
habs geflashed und bekomme kein WLAN...

**Reply by Agerter** [reply]
jetzt mit dem Onlinetool geflashed und es ist WLAN da
![flash](https://github.com/user-attachments/assets/072ef28a-ad4d-4e1f-a50c-a3053e9dc3f6)

**Reply by Manos1966** [reply]
Danke, die Bilder helfen.

Nur damit ich es richtig verstanden habe: Du versuchst diese (roter Kasten) Datei auf eine ESP32-S3 zu flashen und wunderst du warum sie nicht funktioniert?
![Untitled](https://github.com/user-attachments/assets/4b64ae57-4c9d-4484-bc53-d26005a60e40)

**Reply by Agerter** [reply]
ÖÖÖhhh ja ??? ist das falsch.? das habe ich auf die ESP32 immer geflashed...

wer lesen kann ist im Vorteil da gibts ja auch eine S3 Version. ohhhhhhhh mannnnn steinigt mich
sorry

**Reply by Manos1966** [reply]
Kein Problem - möge dass alle deine Probleme so einfach zu lösen sind 😉 

PS. Ganz wichtig: Die opendtu-generic.factory.bin kommt mit PIN Konfiguration.
Bei allen anderen .bin MUSS DU eine PIN Konfiguration selber laden.

**Reply by Agerter** [reply]
danke hat geklappt..WLAN ist da..
kannst du mir noch sagen welche factory bin beim ersten mal flashen nehemn muss , bzw worin liegen die unterschiede in den einzelnen Versionen zb USB Version
![esp](https://github.com/user-attachments/assets/8c84dfd9-d64d-4f26-8ddb-7f26718b33fa)

**Reply by Manos1966** [reply]
![Overview PRe-Compiled Binaries STD OpenDTU](https://github.com/user-attachments/assets/1371119a-e888-4637-ac77-31068db7b663)

**Reply by Agerter** [reply]
danke für die Info... ok und wie erkenne ich ob ich eine UART bridge habe oder integrated USB... ich bin blutiger Anfaenger sorry.

hab die ESP32-S3 mit den beine USB-C einmal steht usb und einmal com drauf

**Reply by Agerter** [reply]
https://www.aliexpress.com/item/1005007148958189.html?spm=a2g0o.order_list.order_list_main.5.4e1818024vfpsN
den chip hab ich gekauft

**Reply by Manos1966** [reply]
Dein Board hat beide.
Auf der Unterseite siehst du COM und USB

Da aber du das flashprogram ESP Downloadtool benutzts, du hast bereits den Virtuellen Treiber USB to COM auf dein PC installiert (COM66) daher macht es keinen Unterschied (so fern ich getestet habe) welches Port du benutzts zum flashen.

**Reply by Manos1966** [reply]
@Agerter Du hast Recht, die Erklärungen reichen nicht aus, ich werde das Wiki ergänzen.

Also, das Board hat beide Ports (USB und USB-to-ART).
![ESP32-S3-DevKitC-1_v2-annotated-photo USB-to-UART and USB b](https://github.com/user-attachments/assets/60d58035-9b29-4b9d-b496-a74a970fbde8)

Grundregel: Du nimmst immer die mit *_usb* am Ende des Namens des Environments (egal ob mit Kabel oder mit OTA). 

- **First Flash:**
  - Man nehme die *opendtu-onbattery-generic_esp32s3_usb.factory.bin* (For boards with integrated usb connection)
  - Mit Kabel flashen kann man die opendtu-onbattery-generic_esp32s3_usb.factory.bin **über beide Ports.** 
  
` Note für die Profis: Falls du am Port mit der USB-to-UART Bridge verbunden bist, nach dem Starten kommt keine Ausgabe, weil die Ausgaben der Konsole auf dem nativen USB-Port landen.`

- **OTA Update:**
  - Man nehme die *opendtu-onbattery-generic_esp32s3_usb.bin* (For boards with integrated usb connection)



` Note für die Profis: Die Variante opendtu-onbattery-generic_esp32s3.factory.bin / opendtu-generic_esp32s3.bin (For boards with usb-uart bridge)  schreibt weiter Nachrichten per UART, und benutzt dafür einen Hardware-UART, und diese Daten werden dann von der UART-Bridge auf USB übersetzt.`

**Comment by jpbuxde** [comment]
Moin,
dazu hätte ich auch noch eine Frage.
Ist es egal ob ich für die Stromversorgung des ESP32S3 den USB oder den COM port verwende?

**Reply by SW-Niko** [reply]
Ich kann jetzt keine Aussage für alle Hersteller geben aber üblicherweise sind die 5V die über die USB/COM ports kommen über Dioden gekoppelt. Dann ist es für den ESP32 egal woher die Energie kommt. Auch beide gleichzeitig ist kein Problem.
Schau mal auf deinen Schaltplan. Wenn du bei beiden Ports eine Diode (Hier D1) findest dann ist alles Ok.
![grafik](https://github.com/user-attachments/assets/1dc80c73-d787-49ab-91ba-9f7af730029e)

**Comment by Harisjems** [comment]
Hallo zusammen, ist es für die Zukunft auch ESP32-C6 N16  Supports eingeplant?
Ich habe hier zwei Sticks zum Spielen übrig.
Oder gibt's da schon was zum Experimentiren.
![Screenshot_20240713_141942_Samsung Internet](https://github.com/user-attachments/assets/3f96d703-66de-4655-8249-c654309f14a9)

**Comment by edddymax** [comment]
Habe eine SolarFlow mit Hub 2000 und HMS 1600. Bisher mit einem ESP32 (openDTUonBattery) verbunden. Alles prima. Ist es sinnvoll/notwendig auf ESP32 - S3 umzustellen? Habe einen ESP32-S3 (ohne Bridge) geflasht (18.08.24) und für die Nutzung soweit vorbereitet. Als Stromzähler habe ich einen Shelly pro 3em eingesetzt. Dabei gab/gibt es ein Problem. Alle Angaben richtig eingestellt, aber beim Aufruf kommt  immer eine http… Fehlermeldung. In der „Live“ - Ansicht wird aber der richtige Wert angezeigt. Wieso ? (In vorheriger Version war „testen“).
